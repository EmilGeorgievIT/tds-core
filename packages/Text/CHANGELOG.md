# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.0.4](https://github.com/telusdigital/tds/compare/@tds/core-text@2.0.3...@tds/core-text@2.0.4) (2019-09-06)

**Note:** Version bump only for package @tds/core-text





## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-text@2.0.2...@tds/core-text@2.0.3) (2019-08-22)

**Note:** Version bump only for package @tds/core-text





## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-text@2.0.1...@tds/core-text@2.0.2) (2019-08-09)

**Note:** Version bump only for package @tds/core-text





## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-text@2.0.0...@tds/core-text@2.0.1) (2019-08-01)


### Bug Fixes

* **core-text:** replace sass with SC in TextSup ([becfe34](https://github.com/telusdigital/tds/commit/becfe34))





# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-text@1.1.0...@tds/core-text@2.0.0) (2019-07-10)


### Features

* **core-text:** upgrade to styled-components ([e9cc4dd](https://github.com/telusdigital/tds/commit/e9cc4dd))


### BREAKING CHANGES

* **core-text:** upgrade to styled-components





# [1.1.0](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.11...@tds/core-text@1.1.0) (2019-06-04)

### Features

- **core-text:** add sup tag support ([e4df9d6](https://github.com/telusdigital/tds/commit/e4df9d6)), closes [#1059](https://github.com/telusdigital/tds/issues/1059)

## [1.0.11](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.10...@tds/core-text@1.0.11) (2019-05-01)

**Note:** Version bump only for package @tds/core-text

## [1.0.10](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.9...@tds/core-text@1.0.10) (2019-03-28)

**Note:** Version bump only for package @tds/core-text

## [1.0.9](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.8...@tds/core-text@1.0.9) (2019-03-20)

**Note:** Version bump only for package @tds/core-text

## [1.0.8](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.7...@tds/core-text@1.0.8) (2019-03-20)

**Note:** Version bump only for package @tds/core-text

## [1.0.7](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.6...@tds/core-text@1.0.7) (2019-03-18)

**Note:** Version bump only for package @tds/core-text

## [1.0.6](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.5...@tds/core-text@1.0.6) (2019-03-05)

**Note:** Version bump only for package @tds/core-text

## [1.0.5](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.4...@tds/core-text@1.0.5) (2019-02-06)

**Note:** Version bump only for package @tds/core-text

<a name="1.0.4"></a>

## [1.0.4](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.3...@tds/core-text@1.0.4) (2018-12-19)

### Bug Fixes

- **core-text:** modify block prop behaviour ([c2f37a6](https://github.com/telusdigital/tds/commit/c2f37a6))

<a name="1.0.3"></a>

## [1.0.3](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.2...@tds/core-text@1.0.3) (2018-12-06)

**Note:** Version bump only for package @tds/core-text

<a name="1.0.2"></a>

## [1.0.2](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.1...@tds/core-text@1.0.2) (2018-07-10)

**Note:** Version bump only for package @tds/core-text

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-text@1.0.0...@tds/core-text@1.0.1) (2018-06-14)

**Note:** Version bump only for package @tds/core-text
