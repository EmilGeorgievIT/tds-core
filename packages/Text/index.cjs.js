require('./dist/index.css')
const Text = require('./dist/index.cjs')

module.exports = Text
