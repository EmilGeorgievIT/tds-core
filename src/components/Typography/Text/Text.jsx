import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import safeRest from '../../../safeRest'

import styles from './Text.modules.scss'

const Text = ({ bold, size, children, ...rest }) => {
  const classes = classnames(
    styles.color,
    styles[size],
    bold ? styles.boldFont : styles[`${size}Font`]
  )

  return (
    <span {...safeRest(rest)} className={classes}>
      {children}
    </span>
  )
}

Text.propTypes = {
  bold: PropTypes.bool,
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large'
  ]),
  children: PropTypes.node.isRequired
}

Text.defaultProps = {
  bold: false,
  size: 'medium'
}

export default Text
