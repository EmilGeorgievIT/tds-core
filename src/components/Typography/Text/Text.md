```
<Text>Tell us what you think</Text>
```

By default, the Text component will inherit font properties from its parent. Use the props to override this behaviour.

```
<Paragraph>
  <Text bold>
    Get Optik TV<Text.Sup>Â®</Text.Sup> and Internet for $85 per month for 12 months.
  </Text>
  <br />
  <Text size="small">
    Sign up for 2 years and save BIG on your first 12 months.
  </Text>
</Paragraph>
```
