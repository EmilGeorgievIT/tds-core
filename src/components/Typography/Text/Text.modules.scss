@import '../../../scss/settings/colours';
@import '../../../scss/settings/typography';

.baseSupSubScripts {
  position: relative;
  vertical-align: baseline;
  padding-left: 0.1em;
  font-size: .875rem;
}

.sup {
  composes: baseSupSubScripts;
  top: -.5em;
}

.sub {
  composes: baseSupSubScripts;
  bottom: -.5em;
}

.base {
  font-size: inherit;
}

.baseFont {
  font-weight: inherit;
}

.small {
  font-size: .875rem;
  letter-spacing: -0.6px;
  line-height: 1.5;
}

.smallFont {
  @include helvetica-neue-roman-55;
}

.medium {
  font-size: 1rem;
  letter-spacing: -0.8px;
  line-height: 1.5;
}

.mediumFont {
  @include helvetica-neue-light-45;
}

.large {
  font-size: 1.25rem;
  letter-spacing: -1px;
  line-height: 2;
}

.largeFont {
  @include helvetica-neue-light-45;
}

.boldFont {
  @include helvetica-neue-medium-65;
}

.color {
  color: $color-text;
}

.colorInverted {
  color: $color-white;
}
