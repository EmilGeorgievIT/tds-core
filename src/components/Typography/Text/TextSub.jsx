import React from 'react'
import PropTypes from 'prop-types'

import styles from './Text.modules.scss'


/**
 * Used for text that should be displayed lower than the main text.
 */
const TextSub = ({ children }) => (
  <sub className={styles.sub}>{children}</sub>
)

TextSub.propTypes = {

  /**
   * Text
   */
  children: PropTypes.node.isRequired
}

TextSub.displayName = 'Text.Sub'

export default TextSub
