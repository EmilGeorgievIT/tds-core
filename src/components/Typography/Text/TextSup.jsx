import React from 'react'
import PropTypes from 'prop-types'

import styles from './Text.modules.scss'


/**
 * Used for text that should be displayed higher than the main text.
 */
const TextSup = ({ children }) => (
  <sup className={styles.sup}>{children}</sup>
)

TextSup.propTypes = {

  /**
   * Superscript text
   */
  children: PropTypes.node.isRequired
}

TextSup.displayName = 'Text.Sup'

export default TextSup
