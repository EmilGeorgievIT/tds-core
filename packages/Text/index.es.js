import './dist/index.css'
import Text from './dist/index.es'

export default Text
