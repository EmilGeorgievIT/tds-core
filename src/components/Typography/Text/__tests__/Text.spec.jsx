import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Text from '../Text'

describe('Text', () => {
  const doShallow = props => (
    shallow(<Text {...props}>Some content</Text>)
  )

  it('renders', () => {
    const text = doShallow()

    expect(toJson(text)).toMatchSnapshot()
  })

  it('renders an HTML span tag', () => {
    const text = doShallow()

    expect(text).toHaveTagName('span')
  })

  it('can be bold', () => {
    let text = doShallow()
    expect(text).not.toHaveClassName('boldFont')
    expect(text).toHaveClassName('mediumFont')

    text = doShallow({ bold: true })
    expect(text).toHaveClassName('boldFont')
    expect(text).not.toHaveClassName('mediumFont')
  })

  it('can be sized', () => {
    let text = doShallow()
    expect(text).toHaveClassName('medium mediumFont')

    text = doShallow({ size: 'small' })
    expect(text).toHaveClassName('small smallFont')
  })

  it('passes additional attributes to the span element', () => {
    const text = doShallow({ id: 'the-text' })

    expect(text).toHaveProp('id', 'the-text')
  })

  it('does not allow custom CSS', () => {
    const text = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(text).not.toHaveProp('className', 'my-custom-class')
    expect(text).not.toHaveProp('style')
  })
})
