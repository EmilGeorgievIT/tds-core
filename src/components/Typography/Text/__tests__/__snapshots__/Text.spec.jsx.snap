// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders 1`] = `
<span
  className="color medium mediumFont"
>
  Some content
</span>
`;
